tests/stress/km/stress_tests_km.cpp

```diff
@@ -639,43 +664,13 @@ _mt_prog_load_stress_test(ebpf_execution_type_t program_type, const test_control
             thread_entry = std::move(std::thread(_test_thread_function, std::ref(context_entry)));
         }
 
+        // If requested, start the 'extension stop-and-restart' thread for extension for this program type.
         if (test_control_info.extension_restart_enabled) {
-
-            // If requested, start the 'extension stop-and-restart' thread for extension for this program type.
-            extension_restart_thread_table.push_back(std::thread([&]() {
-                // Delay the start of this thread for a bit to allow the ebpf programs to attach successfully. There's a
-                // window where if the extension is unloading/unloaded, an incoming attach might fail.
-                std::this_thread::sleep_for(std::chrono::seconds(3));
-
-                // Bump up the priority of this thread so it doesn't get bogged down by the test threads.
-                if (!SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_HIGHEST)) {
-                    auto error = GetLastError();
-                    LOG_WARN(
-                        "WARNING:{} - Failed to increase 'extension restart' thread priority. Error: {}",
-                        __func__,
-                        error);
-                }
-
-                using sc = std::chrono::steady_clock;
-                auto endtime = sc::now() + std::chrono::minutes(test_control_info.duration);
-                while (sc::now() < endtime) {
-
-                    // Drivers can sometimes take some time to stop and (re)start and we need to poll until we can
-                    // determine the final status. 10 (ten) seconds seems a reasonable time for this polling.
-                    constexpr uint32_t RESTART_TIMEOUT_SECONDS = 10;
-                    LOG_VERBOSE("Toggling extension state for {} extension...", program_attribs.extension_name);
-                    if (!_restart_extension(program_attribs.extension_name, RESTART_TIMEOUT_SECONDS)) {
-                        exit(-1);
-                    }
-
-                    LOG_VERBOSE(
-                        "Next restart for {} extension after a delay of {} ms",
-                        program_attribs.extension_name,
-                        test_control_info.extension_restart_delay);
-                    std::this_thread::sleep_for(std::chrono::milliseconds(test_control_info.extension_restart_delay));
-                }
-                LOG_INFO("**** Extension restart thread done. Exiting. ****");
-            }));
+            auto restart_thread = _start_extension_restart_thread(
+                std::ref(program_attribs.extension_name),
+                test_control_info.extension_restart_delay,
+                test_control_info.duration);
```

Comment:
Same comments here about understandability... rename fields so the units are part of the field name.
(BTW that's part of Windows API design guidelines, not unique to ebpf)

Reply:
fixed